rs/nervous_system/integration_tests/src/pocket_ic_helpers.rs

```diff
@@ -1486,38 +1483,27 @@ pub mod sns {
 
         pub async fn propose_and_wait(
             pocket_ic: &PocketIc,
-            canister_id: PrincipalId,
+            governance: GovernanceCanister,
```

Comment:
The pocket_ic_helpers have the advantage that they don't require knowledge of the ic_nervous_system_agent, and therefore can be used in any test that has pocket IC... This adds an additional knowledge dependency to tests that want to take advantage of the pocket ic helpers.  

The simple APIs that these helpers have is a big advantage in being able to compose them together into more complex methods, b/c they are using primitive system types and API types.

I don't like adding the ic_nervous_system_agent as a requirement to use the pocket_ic_helpers.  It seems like we are then mixing up our levels of abstraction... these methods in this file are not meant to be abstracted away from system primitives...

Reply:
That's fair enough! I've reverted that part of this PR